src/fidesops/schemas/connection_configuration/connection_secrets_saas.py

```diff
@@ -49,10 +51,13 @@ def __init__(self, saas_config: SaaSConfig):
     # Pydantic uses the shorthand of (str, ...) to denote a required field of type str
     def get_saas_schema(self) -> Type[SaaSSchema]:
         """Returns the schema for the current configuration"""
-        field_definitions: Dict[str, Any] = {
-            connector_param.name: (str, ...)
-            for connector_param in self.saas_config.connector_params
-        }
+        field_definitions: Dict[str, Any] = {}
```

Comment:
The `field_definitions` are used to denote a parameter as optional or required. A parameter with a default value will be flagged as required and a parameter using Pydantic's `(str, ...)` notation will be flagged as required.

As stated in the comments
```
Pydantic uses the shorthand of (str, ...) to denote a required field of type str
```

Reply:
This comes down to Pydantic's `create_model` function https://pydantic-docs.helpmanual.io/usage/models/#dynamic-model-creation. We need to create a dynamic model for secret validation based on the connector params. A sample `connector_params` to `field_definitions` conversion would look like this:

```
connector_params:
  - name: domain
    default_value: localhost
  - name: username
```
```
{
  "domain": "localhost"
  "username": (str, ...)
}
```
Then the `cls.__fields__` value on line 23 would be
```
{
  'domain': ModelField(name='domain', type=str, required=False, default='localhost'),
  'username': ModelField(name='username', type=str, required=True)
}
```
And we can use the `required` param to determine which secrets will be required by the user.

Reply:
thanks for explaining this, it makes sense now.

i think one thing that tripped me up a bit was a typo in your original comment here - you wrote
> A parameter with a default value will be flagged as required

but instead it should read:
> A parameter with a default value will be flagged as optional

assuming your later comment here is correct. right?

Reply:
You're right, I fixed my typo in case we come back to this.